gobblin-utility/src/main/java/gobblin/util/EmailSender.java

```diff
@@ -0,0 +1,150 @@
+/*
+ *
+ *  * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ *  * this file except in compliance with the License. You may obtain a copy of the
+ *  * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software distributed
+ *  * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ *  * CONDITIONS OF ANY KIND, either express or implied.
+ *
+ */
+
+package gobblin.util;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.password.PasswordManager;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Splitter;
+import com.typesafe.config.Config;
+import org.apache.commons.mail.Email;
+import org.apache.commons.mail.EmailException;
+import org.apache.commons.mail.SimpleEmail;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A helper class for sending emails.
+ */
+public class EmailSender {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(EmailSender.class);
+
+  private final String emailHost;
+  private final Optional<Integer> smtpPort;
+  private final String emailFrom;
+  private final Iterable<String> emailTo;
+  private final boolean needsAuth;
+  private final String emailUser;
+  private final String emailUserPassword;
+  private final PasswordManager passwordManager;
+
+  public EmailSender(Config config) {
+    emailHost = config.hasPath(ConfigurationKeys.EMAIL_HOST_KEY)
+            ?config.getString(ConfigurationKeys.EMAIL_HOST_KEY)
+            :ConfigurationKeys.DEFAULT_EMAIL_HOST;
+
+    smtpPort = config.hasPath(ConfigurationKeys.EMAIL_SMTP_PORT_KEY)
+            ? Optional.of(config.getInt(ConfigurationKeys.EMAIL_SMTP_PORT_KEY))
+            : Optional.<Integer>absent();
+    emailFrom = config.getString(ConfigurationKeys.EMAIL_FROM_KEY);
```

Comment:
If EMAIL_FROM_KEY is absent, config.getString(EMAIL_FROM_KEY) will throw exception